examples/basic_cog.py

```diff
@@ -0,0 +1,115 @@
+    # instead of the bot.command decorator.
+    # They must also pass a `self` parameter since it's within a class.
+    @commands.command()
+    async def hello(self, ctx):
+        """Adds two numbers together"""
+        await ctx.send("Hello am I an example bot!")
+
+    # A command group within a Cog.
+    @commands.group(invoke_without_command=True)
+    async def afk(self, ctx, *, reason: str):
+        """Set your afk status"""
+        user_id = ctx.author.id
+        if user_id not in self.afk_reasons:
+            self.afk_reasons[user_id] = reason
+            mentions = discord.AllowedMentions.none()
+            await ctx.send(f"I've set your afk status to `{reason}`", allowed_mentions=mentions)
+        else:
+            await ctx.send("You are already afk.")
+
+    # Subcommand for the group
+    @afk.command(name='user')
+    async def _user(self, ctx, *, user: discord.User):
+        """Check a user's afk status"""
+        # If a user isn't found an error will be thrown Handle that in an error handler.
+        # If the user input isn't present `MissingRequiredArgument` will be thrown
+        # If a value that cannot be parsed to a member is present, `UserNotFound` will be thrown.
+        # Look at the error handling example
+        mentions = discord.AllowedMentions.none()
+        try:
+            afk = self.afk_reasons[user.id]
+            return await ctx.send(f"{user.name} is afk with reason `{afk}`", allowed_mentions=mentions)
+        except KeyError:
+            return await ctx.send(f'{user.name} is currently not afk', allowed_mentions=mentions)
+
+    # Listening to events within a cog requires the `commands.Cog.listener` decorator.
+    # Note that `self` has to be passed for these as well.
+    @commands.Cog.listener()
+    async def on_message(self, message):
+        # Don't want to reply to bots
+        if message.author.bot:
+            return
+        # Ensure that the message wasn't the afk command
+        ctx = await self.bot.get_context(message)
+        if ctx.valid:
+            return
+
+        # If an afk user has sent a message, remove their afk
+        message_id = message.author.id
+        afk_status = self.afk_reasons.get(message_id)
+        if afk_status:
+            await message.channel.send(f"Welcome back {message.author.mention}! I have removed your afk status.")
+            self.afk_reasons.pop(message_id)
+            return
+
+        # If an afk user is mentioned in a message, let the author of the message know they are afk
+        mentions = {user.id: user for user in message.mentions}
+        mentioned_afk_users = [user_id for user_id in self.afk_reasons.keys() if user_id in mentions]
+        if mentioned_afk_users:
+            plural_or_singular = "is" if len(mentioned_afk_users) == 1 else "are"
+            akf_users = ",".join(str(self.bot.get_user(c)) for c in mentioned_afk_users)
+            await message.channel.send(f"{akf_users} {plural_or_singular} afk.")
```

Comment:
```suggestion
        mentioned_afk_users = [user for user in message.mentions if user.id in self.afk_reasons]
        if mentioned_afk_users:
            plural_or_singular = "is" if len(mentioned_afk_users) == 1 else "are"
            # formats each user as Username#0000
            formatted_users = [str(user) for user in mentioned_afk_users]
            afk_users = ",".join(formatted_users)
            await message.channel.send(f"{afk_users} {plural_or_singular} afk.")
```

This seems better than needing to `get_user` when we already have user object references from the mentions.